packages/loader/container-loader/src/utils.ts

```diff
@@ -23,9 +23,22 @@ export interface ISnapshotTreeWithBlobContents extends ISnapshotTree {
 	trees: { [path: string]: ISnapshotTreeWithBlobContents };
 }
 
+/**
+ * Interface to represent the parsed parts of IResolvedUrl.url to help
+ * in getting info about different parts of the url.
```

Comment:
I'd add
```suggestion
 * in getting info about different parts of the url.
 * May not be compatible or relevant for any Url Resolver
```

It doesn't feel good but I think it's better to be honest.  Maybe we should open a bug to document these hidden requirements of Url Resolver to work with this code in the loader?

Reply:
This is just a temporary change and will go away once the apps own the URL resolvers and the routing too. Also, we are not changing anything, previously also the URL resolvers need to comply to these requirements, otherwise we would fail the load.